pkg/actuators/client/client.go

```diff
@@ -20,86 +20,419 @@ import (
 	"fmt"
 
 	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/platform-services-go-sdk/resourcecontrollerv2"
+	"github.com/IBM/platform-services-go-sdk/resourcemanagerv2"
 	"github.com/IBM/vpc-go-sdk/vpcv1"
+	ibmcloudproviderv1 "github.com/openshift/cluster-api-provider-ibmcloud/pkg/apis/ibmcloudprovider/v1beta1"
 )
 
 // Client is a wrapper object for IBM SDK clients
 type Client interface {
-	InstanceGet(instanceID string) (*vpcv1.Instance, error)
-	// InstanceListAll()
-	// InstanceCreate()
-	// InstanceDelete()
-	// InstanceUpdate()
-	CustomImageGetID(imageName string, regionName string) (*vpcv1.Image, error)
+	// Instances functions
+	InstanceGetByID(instanceID string) (*vpcv1.Instance, error)
+	InstanceExistsByName(name string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec) (bool, error)
+	InstanceGetByName(name string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec) (*vpcv1.Instance, error)
+	InstanceDeleteByName(name string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec) error
+	InstanceCreate(machineName string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec, userData string) (*vpcv1.Instance, error)
+
+	// Helper functions
+	GetCustomImageByName(imageName string, resourceGroupID string) (string, error)
+	GetVPCIDByName(vpcName string, resourceGroupID string) (string, error)
+	GetResourceGroupIDByName(resourceGroupName string) (string, error)
+	GetSubnetIDbyName(subnetName string, resourceGroupID string) (string, error)
+	GetSecurityGroupsByName(securityGroupNames []string, resourceGroupID string, vpcID string) ([]vpcv1.SecurityGroupIdentityIntf, error)
 }
 
 // ibmCloudClient makes call to IBM Cloud APIs
 type ibmCloudClient struct {
-	vpcService *vpcv1.VpcV1
+	vpcService                *vpcv1.VpcV1
+	resourceManagerService    *resourcemanagerv2.ResourceManagerV2
+	resourceControllerService *resourcecontrollerv2.ResourceControllerV2
```

Comment:
Doesn't look as though `resourceControllerService` is being used anywhere in the current code base.
Is it needed?

Reply:
Yup, not anymore, I left it in there in case we need to use it later, but I guess we can add it then.

I was using it to pull a list of all the resources instances in a resource group.